api/tests/Feature/CustomField/CustomFieldTest.php

```diff
@@ -0,0 +1,223 @@
+        $user = factory(User::class)->create();
+        $eventType = factory(EventType::class)->create(['owner_id' => $user->id]);
+
+        $this
+            ->actingAs($user)
+            ->json(
+                'POST',
+                self::EVENT_TYPES_API_URL . '/'
+                . $eventType->id .
+                '/custom-fields',
+                ['custom_fields' => self::CUSTOM_FIELDS_INVALID_TYPE]
+            )
+            ->assertStatus(JsonResponse::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    public function test_add_event_type_with_custom_field_invalid_request_types()
+    {
+        $user = factory(User::class)->create();
+        $eventType = factory(EventType::class)->create(['owner_id' => $user->id]);
+
+        $this
+            ->actingAs($user)
+            ->json(
+                'POST',
+                self::EVENT_TYPES_API_URL . '/'
+                . $eventType->id .
+                '/custom-fields',
+                ['custom_fields' => self::CUSTOM_FIELDS_WRONG_REQUEST_FIELDS_TYPES]
+            )
+            ->assertStatus(JsonResponse::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    public function test_add_event_with_custom_field_values_success()
+    {
+        $user = factory(User::class)->create();
+        $eventType = new EventType([
+            'owner_id' => $user->id,
+            'name' => 'EventType',
+            'description' => '',
+            'slug' => 'event-type-slug',
+            'color' => 'red',
+            'duration' => 60,
+            'timezone' => 'Europe/Kiev',
+            'disabled' => true,
+        ]);
+        $eventType->save();
+
+        $availability = new Availability();
+        $availability->event_type_id = $eventType->id;
+        $availability->type = 'date_range';
+        $availability->start_date = '2020-08-07 11:00:00';
+        $availability->end_date = '2020-08-10 22:00:00';
+        $availability->save();
+
+        $response = $this
+            ->actingAs($user)
+            ->json(
+                'POST',
+                self::EVENT_API_URL,
+                $this->returnEventDataByEventTypeValid($eventType)
+            );
+
+        $response
+            ->assertStatus(JsonResponse::HTTP_NO_CONTENT);
+    }
+
+    public function test_add_event_with_custom_field_values_invalid()
+    {
+        $user = factory(User::class)->create();
+        $eventType = new EventType([
+            'owner_id' => $user->id,
+            'name' => 'EventType',
+            'description' => '',
+            'slug' => 'event-type-slug',
+            'color' => 'red',
+            'duration' => 60,
+            'timezone' => 'Europe/Kiev',
+            'disabled' => true,
+        ]);
+        $eventType->save();
+
+        $availability = new Availability();
+        $availability->event_type_id = $eventType->id;
+        $availability->type = 'date_range';
+        $availability->start_date = '2020-08-07 11:00:00';
+        $availability->end_date = '2020-08-10 22:00:00';
```

Comment:
Could you set up a relative date? If we have correct validation, after a few months the test will not pass.

Reply:
Why it shouldn't pass?

Reply:
Hmm, it is already past date. My thought was that the past date should not be accepted because such an event type will not be available. But a user also should be able to edit old event type, so it should be ok then.